pkg/syncer/volume_health.go

```diff
@@ -77,58 +77,82 @@ func csiGetVolumeHealthStatus(ctx context.Context, k8sclient clientset.Interface
 		}
 	}
 
-	for _, vol := range queryResult.Volumes {
-		log.Debugf("Volume %q Health Status %q", vol.VolumeId.Id, vol.HealthStatus)
+	// volumeIdToHealthStatusMap maps vol.VolumeId.Id to vol.HealthStatus
+	volumeIdToHealthStatusMap := make(volumeIdHealthStatusMap, len(queryResult.Volumes))
 
-		if pvc, ok := volumeHandleToPvcMap[vol.VolumeId.Id]; ok {
-			log.Debugf("csiGetVolumeHealthStatus: Found pvc %q for volume %q", pvc, vol.VolumeId.Id)
+	for _, vol := range queryResult.Volumes {
+		volumeIdToHealthStatusMap[vol.VolumeId.Id] = vol.HealthStatus
+		log.Debugf("csiGetVolumeHealthStatus: Volume Id %s has HealthStatus %s",
+			vol.VolumeId.Id, vol.HealthStatus)
+	}
 
+	for volID, pvc := range volumeHandleToPvcMap {
+		if volHealthStatus, ok := volumeIdToHealthStatusMap[volID]; ok {
+			log.Debugf("csiGetVolumeHealthStatus: Found health status %q for pvc %q, volumeID %q", volHealthStatus, pvc, volID)
```

Comment:
I would also suggest to remove this log as this will appear for all the volumes.

Reply:
Done